crates/ruff_linter/src/rules/ruff/rules/dummy_variable_accessed.rs

```diff
@@ -0,0 +1,213 @@
+    if binding.is_unused() {
+        return None;
+    }
+    // Only variables defined via function arguments or assignments.
+    if !matches!(
+        binding.kind,
+        BindingKind::Argument | BindingKind::Assignment
+    ) {
+        return None;
+    }
+    // This excludes `global` and `nonlocal` variables.
+    if binding.is_global() || binding.is_nonlocal() {
+        return None;
+    }
+
+    let semantic = checker.semantic();
+
+    // Only variables defined in function scopes
+    let scope = &semantic.scopes[binding.scope];
+    if !scope.kind.is_function() {
+        return None;
+    }
+    if !checker.settings.dummy_variable_rgx.is_match(name) {
+        return None;
+    }
+
+    let possible_fix_kind = get_possible_fix_kind(name, checker, binding.scope);
+
+    let mut diagnostic = Diagnostic::new(
+        DummyVariableAccessed {
+            name: name.to_string(),
+            fix_kind: possible_fix_kind,
+        },
+        binding.range(),
+    );
+
+    // If fix available
+    if let Some(fix_kind) = possible_fix_kind {
+        // Get the possible fix based on the scope
+        if let Some(fix) = get_possible_fix(name, fix_kind, binding.scope, semantic) {
+            diagnostic.try_set_fix(|| {
+                Renamer::rename(name, &fix, scope, semantic, checker.stylist())
+                    .map(|(edit, rest)| Fix::safe_edits(edit, rest))
+            });
+        }
+    }
+
+    Some(diagnostic)
+}
+
+/// Enumeration of various ways in which a binding can shadow other variables
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+enum ShadowedKind {
+    /// The variable shadows a global, nonlocal or local symbol
+    Some,
+    /// The variable shadows a builtin symbol
+    BuiltIn,
+    /// The variable shadows a keyword
+    Keyword,
+    /// The variable does not shadow any other symbols
+    None,
+}
+
+/// Suggests a potential alternative name to resolve a shadowing conflict.
+fn get_possible_fix(
+    name: &str,
+    kind: ShadowedKind,
+    scope_id: ScopeId,
+    semantic: &SemanticModel,
+) -> Option<String> {
+    // Remove leading underscores for processing
+    let trimmed_name = name.trim_start_matches('_');
+
+    // Construct the potential fix name based on ShadowedKind
+    let fix_name = match kind {
+        ShadowedKind::Some | ShadowedKind::BuiltIn | ShadowedKind::Keyword => {
+            format!("{trimmed_name}_") // Append an underscore
+        }
+        ShadowedKind::None => trimmed_name.to_string(),
+    };
+
+    // Ensure the fix name is not already taken in the scope or enclosing scopes
+    if !semantic.is_available_in_scope(&fix_name, scope_id) {
+        return None;
+    }
+
+    // Check if the fix name is a valid identifier
+    is_identifier(&fix_name).then_some(fix_name)
+}
+
+/// Determines the kind of shadowing or conflict for a given variable name.
+fn get_possible_fix_kind(name: &str, checker: &Checker, scope_id: ScopeId) -> Option<ShadowedKind> {
```

Comment:
I got a big confused by the terminology `fix_kind` because I assumed it's related to the fix applicability. 

Maybe consider renaming it to `shadowed_kind`? I'd also use `try_` instead of `possible`. `try` is often used for operations that may return `None` or an `Err`.

```suggestion
fn try_shadowed_kind(name: &str, checker: &Checker, scope_id: ScopeId) -> Option<ShadowedKind> {
```